demo/admin/src/products/categories/AssignProductsGrid.tsx

```diff
@@ -0,0 +1,80 @@
+import { gql, useApolloClient, useQuery } from "@apollo/client";
+import { Savable } from "@comet/admin";
+import { CircularProgress } from "@mui/material";
+import {
+    GQLGetProductIdsForProductCategoryQuery,
+    GQLGetProductIdsForProductCategoryQueryVariables,
+    GQLSetProductCategoryMutation,
+    GQLSetProductCategoryMutationVariables,
+} from "@src/products/categories/AssignProductsGrid.generated";
+import { ProductsSelectGrid } from "@src/products/categories/ProductsSelectGrid";
+import isEqual from "lodash.isequal";
+import React, { useEffect, useState } from "react";
+import { FormattedMessage } from "react-intl";
+
+const setProductCategoryMutation = gql`
+    mutation SetProductCategory($id: ID!, $input: [ID!]!) {
+        updateProductCategory(id: $id, input: { products: $input }) {
+            id
+        }
+    }
+`;
+
+const getProductIdsForProductCategory = gql`
+    query GetProductIdsForProductCategory($id: ID!) {
+        products(filter: { category: { equal: $id } }) {
+            nodes {
+                id
+            }
+        }
+    }
+`;
+
+interface FormProps {
+    productCategoryId: string;
+}
+
+export function AssignProductsGrid({ productCategoryId }: FormProps): React.ReactElement {
+    const client = useApolloClient();
+
+    const { data, error, loading } = useQuery<GQLGetProductIdsForProductCategoryQuery, GQLGetProductIdsForProductCategoryQueryVariables>(
+        getProductIdsForProductCategory,
+        {
+            variables: { id: productCategoryId },
+        },
+    );
+    const initialValues = data?.products.nodes ? data.products.nodes.map((product) => product.id).sort() : [];
```

Comment:
Why do you need to sort the array?

Reply:
this is required for `hasChanges={!isEqual(...)}` to work correctly. else deselecting/selecting the same item is detected as change and closing the dialog does ask to save changes even if it hasn't really changed.

Reply:
IMO it would be better to only sort in the hasChanges prop, which would make it more obvious why you sort. Or you could write a custom array comparison and don't use a deep equality check here